doc/design/diagrams/iceoryx_components_diagram_nextgen_release.puml

```diff
@@ -0,0 +1,122 @@
+@startuml
+
+title Eclipse iceoryx Component Overview
+
+skinparam BackgroundColor transparent
+skinparam componentStyle uml2
+
+skinparam component {
+   backgroundColor<<namespace>> yellow
+}
+
+skinparam package {
+   backgroundColor green
+}
+
+skinparam frame {
+   backgroundColor chocolate
+}
+
+' PlantUML package: CMake package
+' PlantUML frame: CMake library
+
+' @todo #590
+' RouDi split, separate files and namespaces for static/dynamic discovery
+' Implement exported libraries (add_library in iceoryx_posh/CMakeLists.txt)
+' RouDiEnvironment shall not be in roudi namespace, but in testing
+
+header Eclipse iceoryx component overview
+
+' HOOFS: Helpful Objects Optimised For Safety
+' Library containing modern C++ STL constructs
+package iceoryx_hoofs  {
+   frame "hoofs library" {
+      component concurrent <<namespace>>
+      component cxx <<namespace>> {
+         component newtype <<namespace>>
+         component set <<namespace>>
+         component algorithm <<namespace>>
+      }
+      component DesignPattern <<namespace>>
+      component errors <<namespace>>
+      component log <<namespace>> {
+         component ffbb <<namespace>>
+      }
+      component units <<namespace>> {
+         component duration_literals <<namespace>>
+
+      }
+      component posix <<namespace>>
+      component threading <<namespace>>
+
+      ' Building blocks which only meet quality level 5
+      ' Can be analysed with tools/analyse_blacklisted_namespaces_in_binary.sh
+      component aux <<namespace>>
+   }
+}
+
+' POSH: POSIX SHared memory
+' Publish/subscribe IPC communication infrastructure based on shared memory
+package iceoryx_posh {
+   frame "core library" {
+      component popo <<namespace>>
+      component capro <<namespace>>
+      component mepoo <<namespace>>
+      component version <<namespace>>
+   }
+
+   frame "runtime library" {
+      component runtime <<namespace>>
+   }
+
+   frame "gateway library" {
+      component gw <<namespace>>
+   }
+
+   frame "config library" {
+      component config <<namespace>>
+      component helper <<namespace>>
+   }
+
+   component errors <<namespace>>
+
+   frame "RouDi library" {
+      component roudi <<namespace>>
+      component build <<namespace>>
```

Comment:
This is not quite correct. It is used in `iceoryx_posh_types.hpp` which is more related to the core library.

Reply:
Moved it.